backend/session/logind.c

```diff
@@ -672,6 +816,27 @@ static struct wlr_session *logind_session_create(struct wl_display *disp) {
 		goto error_bus;
 	}
 
+	// Check for CanGraphical first
+	sd_bus_error error = SD_BUS_ERROR_NULL;
+	ret = sd_bus_get_property_trivial(session->bus,
+		"org.freedesktop.login1", session->seat_path,
+		"org.freedesktop.login1.Seat", "CanGraphical", &error,
+		'b', &session->can_graphical);
+	if (ret < 0) {
+		wlr_log(WLR_ERROR, "Failed to get 'CanGraphical' property: %s",
+			error.message);
+		sd_bus_error_free(&error);
+		goto error_bus;
+	}
+
+	if (!session->can_graphical) {
+		wlr_log(WLR_INFO, "Waiting for 'CanGraphical' on seat %s", session->base.seat);
+	}
+
+	while (!session->can_graphical) {
+		wl_event_loop_dispatch(event_loop, -1);
```

Comment:
This function can return an error. Need to exit when it's the case, instead of entering in a busy loop.

Reply:
Makes sense, thanks. I added an error handling bit.